### PR TITLE
Icemoon

### PR DESCRIPTION
**Pull request by Whoneedspacee**
## About The Pull Request

Adds a couple new ruins and a new megafauna to be the equivalent of blood drunk miner on ice moon. Adds a capability for weather to not target indoor areas, removing the need for an area target on a z level. Areas can now specify if they want megafauna or mobs or tunnels to go into them. There are now new spawners for 2 new mobs on ice moon wolves and polar bears. Tunnels can now have varying widths you can define on tunnel producing type. Ice moon also has a unique planetary atmos like lavaland. Admins can pick which mining map they want to be picked next, otherwise it will be chosen randomly. Ruins can now spawn in multiple area types on the same z level which fixes an issue with multiz ruins. The ice moon budget for spawning ruins is larger due to the multiz making things very spread out.

https://m.youtube.com/watch?v=iB-hkN_wYmY&t=
https://m.youtube.com/watch?v=iB-hkN_wYmY&t=
https://m.youtube.com/watch?v=iB-hkN_wYmY&t=

## Why It's Good For The Game

Variation is fun, this provides a lot of tools to make other additions like this possible as well. This isn't the most finished thing I've put out in terms of new content but I'm hoping the multiz will inspire people to make unique ruins, and I'll probably add some myself over time.

## Changelog
:cl:
add: Ice moon, sometimes spawns in the place of lavaland with a couple different ruins, mobs, and a lower z level
admin: Admins can choose the next mining map in the same tab as normal map choosing
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
thats the way uh huh uh huh i like it

**Comment by itseasytosee** [comment]
Does it have any gib ability? It seems like it would be really easy to cheese with dead bodies

**Comment by Whoneedspacee** [comment]
At the end it gibs the target, works the same as any other megafauna.

**Comment by Shadowflame909** [comment]
How do we get the ash-drakes white, and shooting out whatever chemical is in that syndicate Freon grenade. Glyon? No clue, but the ice grenade of liquid death.

**Comment by Whoneedspacee** [comment]
give me sprites and i will do great things with them

the snowballs are just the nuclear particle sprite this is what im working with

**Comment by cacogen** [comment]
It uses multi-z? So it's going to be buggy as fuck.

**Comment by Farquaar** [comment]
Looks cool! How does this affect the gulag?

**Comment by trollbreeder** [comment]
@Shadowflame909 it's called frost oil i think, it rapidly cools you down, it's also a very potent lizard exterminator

**Comment by necromanceranne** [comment]
Nice, I didn't know you were making it multiz! That's gonna be sweet!

**Comment by Whoneedspacee** [comment]
> It uses multi-z? So it's going to be buggy as fuck.

so we're going to fix the bugs really good 👍

**Comment by Dennok** [comment]
> It uses multi-z? So it's going to be buggy as fuck.

I think it means multi-z ruins, this different thing now. It create ruin room on special z-lvl with ladder to small ruin entrance on mining area.

**Comment by Dorsisdwarf** [comment]
Oh damn, this looks incredible! Lavaland use of multiZ might be a better platform for testing than multistations anyway

**Comment by ATH1909** [comment]
Finally, we can make chasms not insta-kill people (we can just make them send people down a z-level instead).

**Comment by lolman360** [comment]
good pr.

are we gonna get something akin to tendrils here or is there going to be another way to get infinite mobs/special loot?

**Comment by itseasytosee** [comment]
Why do the wolves drop corgi skin and meat?

**Comment by Dennok** [comment]
> good pr.
> 
> are we gonna get something akin to tendrils here or is there going to be another way to get infinite mobs/special loot?

`cave entrance` is Icemoon spawner that can spawns wolfs and bears now.

**Comment by Whoneedspacee** [comment]
> Why do the wolves drop corgi skin and meat?

thanks for pointing out things that were placeholder 👍

**Comment by bobbahbrown** [comment]
Marked WIP per discussion with Whoneedsspace on 2020-02-17.

File: code/modules/mapping/ruins.dm [hunk 1]
```diff
@@ -11,8 +11,15 @@
...
-			if(!(istype(new_area, allowed_areas)) || check.flags_1 & NO_RUINS_1)
+			if(check.flags_1 & NO_RUINS_1)
 				valid = FALSE
+			else
+				for(var/type in allowed_areas)
+					valid = FALSE // set to false before we check
+					if(istype(new_area, type)) // it's at least one of our types so it's whitelisted
+						valid = TRUE
+						break
+			if(!valid)
```
Comment (bobbahbrown) [inline code/modules/mapping/ruins.dm]: you could move the valid = FALSE assignment to outside the loop no need to reassign each time with the break

**Comment by Dennok** [comment]
Maybe make special wall around boss that you show in video?
Special hard rock that can be mined by mastered miner and immune to boss attacks. 
I hate this megafauna like colossus that anihilate all nearby place when someone fight it.

**Comment by IndieanaJones** [comment]
I'm happy to see this is finally here.  It'll take a bit to get the Ice Moon to be as content rich as Lavaland, but I think that'll be interesting in its own right.  Also, if the Ice Moon is selected, does it change the planet in the background visible from the station?

**Comment by Whoneedspacee** [comment]
thats one of the few things i need to change as well as the shuttle sendoff text in the ui there thanks for reminding me

if anyone has that parallax or knows how the hell that shit works please dm me or post here, i know someone made it before

**Comment by IndieanaJones** [comment]
Also, a couple of personal thoughts:
- Polar bears should have something more than just being tendril-less goliaths.  Perhaps they could have a pounce akin to the xeno hunter, where you can to step out of the way in order to not get tackled?
- The miner megafauna should probably not be a miner.  As it stands, the Blood-Drunk miner is sort of a lazy and forgettable entity who barely qualifies as a megafauna, he'd be much more interesting and cool if he was some sort of monster instead
- Ash Walkers should have some equal on the ice moon, but probably not be the same.  Other ghost roles like golems, syndies, and podpeeps can fit in both places, but the ashies are heavily linked to lavaland.  Could literally be the same with a race change, or something to change down the line.

**Comment by Whoneedspacee** [comment]
1. sure

2. ice moon was meant to have an eldritchy background so this is kinda that with the miner being possessed or something, if you have sprites to improve it sure, but i took a sprite and made a boss from it

3. sprites

**Comment by Dennok** [comment]
3. You can use cat peoples

**Comment by FantasticFwoosh** [comment]
Very talented attempt, i've had eyes on a Ice-Moon concept for a while I hope you can take it to new places rather than turn the heat down on Lavaland. Contact me through DM if you need any sprite requests as i have a little cache of concept doodles and time at the moment.

>     1. You can use cat girls
Im not sure they need the encouragement of diversification + resulting drama about seeking approval to do it. Begrudgingly recommending bear people or satirical comedy human russians maybe.

**Comment by trollbreeder** [comment]
territorial russians that run on 19-20th century technology and use a rickidy-dickidy ol' fabricator and they're chaotic neutral, yes i love the idea already

**Comment by fludd12** [comment]
catgirls totally meet the requirement for eldritch theme, though
they can worship...

...

*nya*~rlathothep. 😎

**Comment by PKPenguin321** [comment]
I think @Qustinnus wrote parallax for us? Or at least the lavaland planet part

**Comment by Whoneedspacee** [comment]
if i could have it my way then ice moon would be comprised entirely of new content but if i did that then it would never get merged

if anyone wants to add ruins to this feel free, you can dm me if you want assistance or items added for it

**Comment by Farquaar** [comment]
There's an spear-wielding eskimo sprite in simplehuman.dmi that could probably be used as a placeholder for the miner sprite until someone makes a good looking monster.

**Comment by Valtosin** [comment]
I testmerged this PR, so, there is some issues with it:

1. You can't build/dig ladder/stairs if you fall down. Only one ladder at center.
2. You also can't dig down.
3. Ashwalkers isn't cold-resistant people. They die without boots, even if wintercoat.
3.1 Also campfire doesn't work as heat source.
4. Wolfs and bears are strong.
5. You can fall into ruin's end from above and loot all things without completing it.

**Comment by optimumtact** [comment]
I don't want to carry all of these sprites, code and content for something that shows up every other round.

I'm not a fan of lavaland as it is, as I think it needs to be more firewalled from the station, and there's too much of a focus from people on content that isn't the main station.

This could be a main station map, or a gamemode, but I don't see us adding this as a lavaland "reskin"

**Comment by FancyCyclops** [comment]
Oshan Lab.

**Pull request closed without merging**